pkg/parser/pancakeswap/parser.go

```diff
@@ -0,0 +1,293 @@
+	if err != nil {
+		return storage.TradeLog{}, err
+	}
+
+	order, err = p.detectRfqTrade(order)
+	if err != nil {
+		return order, err
+	}
+	return order, nil
+}
+
+func (p *Parser) buildOrderByLog(log ethereumTypes.Log, blockTime uint64) (storage.TradeLog, error) {
+	if len(log.Topics) > 0 && log.Topics[0].Hex() != p.eventHash {
+		return storage.TradeLog{}, parser.ErrInvalidTopic
+	}
+	e, err := p.ps.ParseFill(log)
+	if err != nil {
+		return storage.TradeLog{}, err
+	}
+	order := storage.TradeLog{
+		OrderHash:       common.Hash(e.OrderHash).String(),
+		ContractAddress: e.Raw.Address.String(),
+		BlockNumber:     e.Raw.BlockNumber,
+		TxHash:          e.Raw.TxHash.String(),
+		LogIndex:        uint64(e.Raw.Index),
+		Timestamp:       blockTime * 1000,
+		EventHash:       p.eventHash,
+		Maker:           e.Filler.String(),
+		Taker:           e.Swapper.String(),
+	}
+	return order, nil
+}
+
+func (p *Parser) detectRfqTrade(order storage.TradeLog) (storage.TradeLog, error) {
+	traceCall, err := p.traceCalls.GetTraceCall(order.TxHash)
+	if err != nil {
+		return order, err
+	}
+
+	order, err = p.recursiveDetectRFQTrades(order, traceCall)
+	if err != nil {
+		return order, err
+	}
+	return order, nil
+}
+
+func (p *Parser) recursiveDetectRFQTrades(order storage.TradeLog, call types.CallFrame) (storage.TradeLog, error) {
+	for _, l := range call.Logs {
+		if len(l.Topics) < 2 {
+			continue
+		}
+		if l.Topics[0].String() != p.eventHash || l.Topics[1].String() != order.OrderHash {
+			continue
+		}
+		input, err := decoder.Decode(p.abi, call.Input)
+		if err != nil {
+			continue
+		}
+		if len(input.Params) == 0 {
+			continue
+		}
+		inputOrder, ok := input.Params[0].Value.(struct {
+			Order []uint8 `json:"order"`
+			Sig   []uint8 `json:"sig"`
+		})
+		if !ok {
+			inputOrders, ok := input.Params[0].Value.([]struct {
+				Order []uint8 `json:"order"`
+				Sig   []uint8 `json:"sig"`
+			})
+			if !ok || len(inputOrders) == 0 {
+				continue
+			}
+			inputOrder = inputOrders[0]
+		}
+		parsedOrder, err := p.orderArguments.Unpack(inputOrder.Order)
+		if err != nil {
+			continue
+		}
+		finalOrder, err := p.updateOrder(order, parsedOrder)
+		if err != nil {
+			continue
+		}
+		return finalOrder, nil
+	}
+
+	for i := range call.Calls {
+		order, err := p.recursiveDetectRFQTrades(order, call.Calls[i])
+		if err == nil {
+			return order, nil
+		}
+	}
+	traceData, _ := json.Marshal(call)
+	return order, fmt.Errorf("%w %s", parser.ErrNotFoundTrade, string(traceData))
+}
+
+func (p *Parser) updateOrder(internal storage.TradeLog, parsed []interface{}) (storage.TradeLog, error) {
+	data, err := json.Marshal(parsed)
+	if err != nil {
+		return storage.TradeLog{}, err
+	}
+	var resolvedOrder []ResolvedOrder
+	if err = json.Unmarshal(data, &resolvedOrder); err != nil {
+		return storage.TradeLog{}, err
+	}
+	if len(resolvedOrder) == 0 || len(resolvedOrder[0].Outputs) == 0 {
+		return storage.TradeLog{}, err
+	}
```

Comment:
should we return this ?
```suggestion
	if len(resolvedOrder) == 0 || len(resolvedOrder[0].Outputs) == 0 {
		return internal, nil
	}
```

Reply:
the original tradelog is returned on the higher level so I think there no need to return it here